server/cluster/manager_test.go

```diff
@@ -56,39 +57,87 @@ func prepareEtcdServerAndClient(t *testing.T) (*embed.Etcd, *clientv3.Client, fu
 	return etcd, client, clean
 }
 
-func newClusterManager(t *testing.T) Manager {
+func newTestStorage(t *testing.T) Manager {
 	_, client, _ := prepareEtcdServerAndClient(t)
-	storage := storage.NewStorageWithEtcdBackend(client, "/", storage.Options{
+	storage := storage.NewStorageWithEtcdBackend(client, "/aaa", storage.Options{
 		MaxScanLimit: 100, MinScanLimit: 10,
 	})
 	return NewManagerImpl(storage, schedule.NewHeartbeatStreams(context.Background()))
 }
 
-func TestManager(t *testing.T) {
+func TestManagerWithSingleThread(t *testing.T) {
 	re := require.New(t)
-	manager := newClusterManager(t)
+	manager := newTestStorage(t)
 
 	ctx := context.Background()
-	testCreateCluster(ctx, re, manager)
+	testCreateCluster(ctx, re, manager, cluster1, false)
+	testCreateCluster(ctx, re, manager, cluster1, true)
 
-	testRegisterNode(ctx, re, manager, defaultCluster, node1, defaultLease)
-	testRegisterNode(ctx, re, manager, defaultCluster, node2, defaultLease)
+	testRegisterNode(ctx, re, manager, cluster1, node1, defaultLease)
+	testRegisterNode(ctx, re, manager, cluster1, node2, defaultLease)
 
-	testAllocSchemaID(ctx, re, manager, defaultCluster, defaultSchema, defaultSchemaID)
+	testAllocSchemaID(ctx, re, manager, cluster1, defaultSchema, defaultSchemaID)
+	testAllocSchemaID(ctx, re, manager, cluster1, defaultSchema, defaultSchemaID)
 
-	testAllocTableID(ctx, re, manager, node1, defaultCluster, defaultSchema, table1, tableID1)
-	testAllocTableID(ctx, re, manager, node1, defaultCluster, defaultSchema, table1, tableID1)
-	testAllocTableID(ctx, re, manager, node1, defaultCluster, defaultSchema, table2, tableID2)
-	testAllocTableID(ctx, re, manager, node2, defaultCluster, defaultSchema, table3, tableID3)
-	testAllocTableID(ctx, re, manager, node2, defaultCluster, defaultSchema, table4, tableID4)
+	testAllocTableID(ctx, re, manager, node1, cluster1, defaultSchema, table1, tableID1)
+	testAllocTableID(ctx, re, manager, node1, cluster1, defaultSchema, table1, tableID1)
+	testAllocTableID(ctx, re, manager, node1, cluster1, defaultSchema, table2, tableID2)
+	testAllocTableID(ctx, re, manager, node2, cluster1, defaultSchema, table3, tableID3)
+	testAllocTableID(ctx, re, manager, node2, cluster1, defaultSchema, table4, tableID4)
 
-	testGetTables(ctx, re, manager, node1, defaultCluster)
-	testGetTables(ctx, re, manager, node2, defaultCluster)
+	testDropTable(ctx, re, manager, cluster1, defaultSchema, table1, tableID1)
+	testDropTable(ctx, re, manager, cluster1, defaultSchema, table3, tableID3)
+
+	testGetTables(ctx, re, manager, node1, cluster1)
+	testGetTables(ctx, re, manager, node2, cluster1)
 }
 
-func testCreateCluster(ctx context.Context, re *require.Assertions, manager Manager) {
-	_, err := manager.CreateCluster(ctx, defaultCluster, defaultNodeCount, defaultReplicationFactor, defaultShardTotal)
-	re.NoError(err)
+func TestManagerWithMultiThread(t *testing.T) {
```

Comment:
TestManagerMultiThread maybe reuse TestManagerSingleThread ?

Reply:
Single-threaded test cases cannot be reused due to concurrent testing within multi-threaded test cases